### PR TITLE
Provide option to cleanup Docker assets

### PR DESCRIPTION
**Pull request by ravimeda**
Refer to https://github.com/dotnet/dotnet-docker-nightly/issues/529

### CODE REVIEW

**Comment by ravimeda** [comment]
@dotnet-bot test ci please

File: netci.groovy [hunk 1]
```diff
@@ -26,7 +26,7 @@ platformList.each { platform ->
         def newJob = job(newJobName) {
             steps {
                 if (hostOS == 'Windows_2016') {
-                    batchFile("powershell -NoProfile -Command .\\build-and-test.ps1 -VersionFilter \"${versionFilter}\" -OSFilter \"${containerOS}\"")
+                    batchFile("powershell -NoProfile -Command .\\build-and-test.ps1 -VersionFilter \"${versionFilter}\" -OSFilter \"${containerOS}\" -CleanupDocker")
                 }
                 else {
                     shell("docker build --rm -t testrunner -f ./test/Dockerfile.linux.testrunner . && docker run -v /var/run/docker.sock:/var/run/docker.sock testrunner powershell -File build-and-test.ps1 -VersionFilter \"${versionFilter}\"")
```
Comment (MichaelSimons) [inline netci.groovy]: We should be invoking the cleanup logic on Linux as well

File: build-and-test.ps1 [hunk 2]
```diff
@@ -3,12 +3,29 @@ param(
...
 
 Set-StrictMode -Version Latest
 $ErrorActionPreference = 'Stop'
 
+function Invoke-CleanupDocker()
+{
+    if ($CleanupDocker)
+    {
+        docker ps -a -q | ForEach-Object { docker rm -f $_ }
+        docker images |
```
Comment (MichaelSimons) [inline build-and-test.ps1]: I think a comment would be justified on why the base Windows OS images are left.

File: build-and-test.ps1 [hunk 3]
```diff
@@ -22,6 +39,7 @@ $manifest = Get-Content "manifest.json" | ConvertFrom-Json
 $manifestRepo = $manifest.Repos[0]
 $activeOS = docker version -f "{{ .Server.Os }}"
 $builtTags = @()
+Invoke-CleanupDocker
```
Comment (MichaelSimons) [inline build-and-test.ps1]: I think cleaning up the environment should be done as soon as possible - e.g. line 30.

**Pull request merged**